src/account_id.rs

```diff
@@ -0,0 +1,412 @@
+        self.as_str().partial_cmp(&other.0)
+    }
+}
+
+impl<'a> PartialOrd<AccountId> for &'a AccountIdRef {
+    fn partial_cmp(&self, other: &AccountId) -> Option<std::cmp::Ordering> {
+        self.0.partial_cmp(other.as_str())
+    }
+}
+
+impl<'a> PartialOrd<&'a AccountIdRef> for AccountId {
+    fn partial_cmp(&self, other: &&'a AccountIdRef) -> Option<std::cmp::Ordering> {
+        self.as_str().partial_cmp(&other.0)
+    }
+}
+
+impl PartialOrd<AccountId> for String {
+    fn partial_cmp(&self, other: &AccountId) -> Option<std::cmp::Ordering> {
+        self.as_str().partial_cmp(other.as_str())
+    }
+}
+
+impl PartialOrd<String> for AccountId {
+    fn partial_cmp(&self, other: &String) -> Option<std::cmp::Ordering> {
+        self.as_str().partial_cmp(other.as_str())
+    }
+}
+
+impl PartialOrd<AccountId> for str {
+    fn partial_cmp(&self, other: &AccountId) -> Option<std::cmp::Ordering> {
+        self.partial_cmp(other.as_str())
+    }
+}
+
+impl PartialOrd<str> for AccountId {
+    fn partial_cmp(&self, other: &str) -> Option<std::cmp::Ordering> {
+        self.as_str().partial_cmp(other)
+    }
+}
+
+impl<'a> PartialOrd<AccountId> for &'a str {
+    fn partial_cmp(&self, other: &AccountId) -> Option<std::cmp::Ordering> {
+        self.partial_cmp(&other.as_str())
+    }
+}
+
+impl<'a> PartialOrd<&'a str> for AccountId {
+    fn partial_cmp(&self, other: &&'a str) -> Option<std::cmp::Ordering> {
+        self.as_str().partial_cmp(*other)
+    }
+}
+
+impl<'a> From<AccountId> for Cow<'a, AccountIdRef> {
+    fn from(value: AccountId) -> Self {
+        Cow::Owned(value)
+    }
+}
+
+impl<'a> From<&'a AccountId> for Cow<'a, AccountIdRef> {
+    fn from(value: &'a AccountId) -> Self {
+        Cow::Borrowed(value)
+    }
+}
+
+impl<'a> From<Cow<'a, AccountIdRef>> for AccountId {
+    fn from(value: Cow<'a, AccountIdRef>) -> Self {
+        value.into_owned()
+    }
+}
+
+#[cfg(feature = "arbitrary")]
+impl<'a> arbitrary::Arbitrary<'a> for AccountId {
+    fn size_hint(_depth: usize) -> (usize, Option<usize>) {
+        (AccountId::MIN_LEN, Some(AccountId::MAX_LEN))
+    }
+
+    fn arbitrary(u: &mut arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        let mut s = u.arbitrary::<&str>()?;
+        loop {
+            match s.parse::<AccountId>() {
+                Ok(account_id) => break Ok(account_id),
+                Err(ParseAccountError {
+                    char: Some((idx, _)),
+                    ..
+                }) => {
+                    s = &s[..idx];
+                    continue;
+                }
+                _ => break Err(arbitrary::Error::IncorrectFormat),
+            }
+        }
+    }
+
+    fn arbitrary_take_rest(u: arbitrary::Unstructured<'a>) -> arbitrary::Result<Self> {
+        <&str as arbitrary::Arbitrary>::arbitrary_take_rest(u)?
+            .parse()
+            .map_err(|_| arbitrary::Error::IncorrectFormat)
+    }
+}
```

Comment:
I wonder if we can re-use AccountIdRef implementation instead of having a copy of it in two places.

Reply:
Probably! I'll give it a try.